### PR TITLE
Added UT/IT

### PR DESCRIPTION
**Pull request by ViKrAm-Bais**

### CODE REVIEW

File: TransporterAPI/pom.xml [hunk 1]
```diff
@@ -92,6 +92,28 @@
...
+</dependency>
+
+
+
+<dependency>
+   <groupId>org.hibernate.validator</groupId>
+   <artifactId>hibernate-validator-cdi</artifactId>
+   <version>7.0.0.Final</version>
+</dependency>
+
```
Comment (sritambehera) [inline TransporterAPI/pom.xml:116]: hibernate dependency is being added twice unnecessarily.

File: TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java [hunk 2]
```diff
@@ -11,12 +11,16 @@
...
+import lombok.AllArgsConstructor;
 import lombok.Data;
+import lombok.NoArgsConstructor;
 
 
 @Data
 @Table(name = "Transporter")
 @Entity
+@AllArgsConstructor
+@NoArgsConstructor
```
Comment (sritambehera) [inline TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:23]: What's the purpose of these two lines?

Comment (ViKrAm-Bais) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:23]: creating constructor, without this annotation it was giving error (to create constructor with all params )

Comment (sritambehera) [reply TransporterAPI/src/main/java/com/springboot/TransporterAPI/Entity/Transporter.java:23]: ok

File: TransporterAPI/src/test/java/com/springboot/TransporterAPI/TestTransporterService.java [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+	@Order(1)
+	public void addTransporterfail_phoneno_null()
+	{
+		PostTransporter posttransporter = new PostTransporter( null, "Nagpur", "transporter1", "company1", "link1");
+		
+		Set<ConstraintViolation<PostTransporter>> constraintViolations = validator.validate(posttransporter);
+	    
+		assertEquals(1, constraintViolations.size());
+		assertEquals("Phone no. cannot be blank!", constraintViolations.iterator().next().getMessage());
+	}
```
Comment (sritambehera) [inline TransporterAPI/src/test/java/com/springboot/TransporterAPI/TestTransporterService.java:100]: check for blank phone no. case like phoneNo="             ";

File: TransporterAPI/src/test/java/com/springboot/TransporterAPI/TestTransporterService.java [hunk 4]
```diff
@@ -0,0 +1,422 @@
...
+	public void getTransporters1()
+	{
+		List<Transporter> transporters = createTransporters();
+		Pageable page = PageRequest.of(0, 15, Sort.Direction.DESC, "timestamp");
+		when(transporterdao.findByTransporterCompanyApproved(false, false, page)).thenReturn(transporters);
+		Collections.reverse(transporters);
+		List<Transporter> transportersres = transporterservice.getTransporters(false, false, null, 0);
+		
+		assertEquals(transporters, transportersres);
+	}
```
Comment (sritambehera) [inline TransporterAPI/src/test/java/com/springboot/TransporterAPI/TestTransporterService.java:255]: no_of_pages +=(lastpage_count+x)/page_size;
last_page_count =(last_page_count+x)%page_size;


Add these two lines for checking the correctness after adding certain no of elements to the returned data.

**Review by liveasy97 — APPROVED** [review]

**Pull request merged**